### ISSUE TITLE
Feature Requirement: coordinate rounding

### ISSUE THREAD

**Issue by truth-quark** [ACCESS-NRI/um2nc-standalone#2]
`um2nc` needs to process UM files which can have issues with coordinate rounding.

`TODO is this correct?` Some floating point coordinates are not exact values, e.g. the coordinate is `1000.000001` instead of `1000.0`. This can be confusing & can prevent some data alignment. Rounding is applied to clean coordinate values.

* Is this mainly in pressure level variables?
* Is this the corresponding code block? https://github.com/ACCESS-NRI/um2nc-standalone/blob/main/src/um2netcdf.py#L48-L86
* Due to numerical precision problem representing floats in a binary? https://docs.python.org/3/tutorial/floatingpoint.html

Requirements:

* What coordinate rounding fixes are required?
* Which data files are affected?
* Locate small data files to use for unit test data
* Add tests for the cases
  - Tests cover `[1.0, 2.0, ...]` -- >  `[1.0, 2.0, ...]` case as a baseline
  - Ensure tests cover `[1.0000001, 2.0000001, ...]` rounding to 0 case
  - Does the tolerance need to be configurable? (then by extension, a cmd line arg)

**Comment by MartinDix** [comment]
import iris
```
ta = iris.load_cube('/g/data/tm70/esm15_testing/sample_output/aiihca.paa1jan',  constraint=iris.AttributeConstraint(STASH='m01s30i204'))
print(ta.coord('pressure').points)
print(ta.coord('pressure').points[-1] - 1000.)

[   1.    5.   10.   20.   30.   50.   70.  100.  150.  200.  250.  300.
  400.  500.  600.  700.  850.  925. 1000.]

1.000444171950221e-10
```

The dimension variables in the netCDF file are 64 bit, so this gets saved as a value different to 1000 which is annoying/confusing and sometimes causes problems comparing with other variables where the level is set exactly.

**Comment by truth-quark** [comment]
Closed by #94.
